autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -268,13 +279,13 @@ def _check_data(
             X = cast(pd.DataFrame, X)
 
             # Handle objects if possible
-            if not X.select_dtypes(include='object').empty:
+            object_columns_indicator = has_object_columns(X.dtypes)
```

Comment:
is this a faster way to test for column with 'object' type?

Reply:
Hmm, I tried to do it in a faster way with the function I added, `has_object_columns` which does not return a ` DataFrame`.

Reply:
no I think its fine for now

Reply:
@nabenabe0928 Thanks for the code, it pointed out the fact that my code was going through all features and not just return if it found the first object one.

How about this implementation

```
    def arlind_v2(X) -> bool:
        feature_types = X.dtypes
        for feature_type in feature_types:
            if pd.api.types.is_object_dtype(feature_type):
                return True
            else:
                return False
```

```
Old version
594 µs ± 14.6 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
1.35 ms ± 9.81 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
Arlind version
914 µs ± 11.5 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
917 µs ± 17.8 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
Shuhei version
107 µs ± 1.03 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
83 µs ± 695 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
Arlind v2 version
57.7 µs ± 684 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
57.9 µs ± 730 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

Reply:
@ArlindKadra Thanks for the suggestion:)
I also thought in the same way because early-return is a good idea, but this implementation only looks at the first element, so you need to do the following:
```
    def arlind_v2(X) -> bool:
        feature_types = X.dtypes
        for feature_type in feature_types:
            if pd.api.types.is_object_dtype(feature_type):
                return True

        return False
```

Reply:
ups, very right, a typo and bug right there.

Reply:
```
Old version
626 µs ± 26.7 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
1.42 ms ± 23.6 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
Arlind version
974 µs ± 5.84 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
966 µs ± 9.93 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
Shuhei version
109 µs ± 2.58 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
84.4 µs ± 980 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
Arlind v2 version
932 µs ± 10.2 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
59.5 µs ± 1.45 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

Does not seem faster anymore, awful when working with numerical values.